File: frontend/src/models/Asset.ts
export type AssetStatus = "Available" | "Assigned" | "Borrowed" | "Reserved" | "Maintenance" | "In Transit";

export class Asset {
  constructor(
    public readonly id: string,
    public readonly name: string,
    public readonly serial: string,
    public readonly category: string,
    public readonly location: string,
    public readonly valueLabel: string,
    public readonly status: AssetStatus,
    public readonly imageUrl: string,
    public readonly rateLabel?: string,
    public readonly custodian?: string,
    public readonly utilization?: number,
  ) {}
}

File: backend/crates/common/src/models/asset.rs
use serde::{Deserialize, Serialize};
use utoipa::ToSchema;

#[derive(Debug, Clone, Serialize, Deserialize, ToSchema)]
#[serde(rename_all = "camelCase")]
pub struct Asset {
    pub id: String,
    pub name: String,
    pub serial_number: String,
    pub image_url: String,
    pub status: AssetStatus,
    pub category: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub custodian_name: Option<String>,
    pub value_usd: f64,
}

#[derive(Debug, Clone, Copy, Serialize, Deserialize, ToSchema, PartialEq, Eq)]
#[serde(rename_all = "lowercase")]
pub enum AssetStatus {
    Assigned,
    Available,
    Maintenance,
}
